patches/server/0948-Optimize-VarInts.patch

```diff
@@ -0,0 +1,52 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: astei <andrew@steinborn.me>
+Date: Sat, 26 Nov 2022 10:56:55 +0100
+Subject: [PATCH] Optimize VarInts
+
+https://github.com/PaperMC/Paper/pull/6957#issuecomment-985250854
+
+diff --git a/src/main/java/net/minecraft/network/FriendlyByteBuf.java b/src/main/java/net/minecraft/network/FriendlyByteBuf.java
+index 35377576ed182814051c11f902e02e8e921e84e3..3c5fe2e792fafe3bd554e5af0f1c2e24c2ca0422 100644
+--- a/src/main/java/net/minecraft/network/FriendlyByteBuf.java
++++ b/src/main/java/net/minecraft/network/FriendlyByteBuf.java
+@@ -88,6 +88,18 @@ public class FriendlyByteBuf extends ByteBuf {
+     }
+ 
+     public static int getVarIntSize(int value) {
++    //Paper start - Optimize VarInts
++        return VARINT_EXACT_BYTE_LENGTHS[Integer.numberOfLeadingZeros(value)]; // Paper - Optimize VarInts
++    }
++    private static final int[] VARINT_EXACT_BYTE_LENGTHS = new int[33];
++    static {
++        for (int i = 0; i <= 32; ++i) {
++            VARINT_EXACT_BYTE_LENGTHS[i] = (int) Math.ceil((31d - (i - 1)) / 7d);
++            VARINT_EXACT_BYTE_LENGTHS[32] = 1; // Special case for the number 0.
```

Comment:
Is there any reason why this is inside the loop?

Reply:
No. Looks like a copy pasta problem. Will fix soon.